core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -70,6 +69,8 @@ public static XContentBuilder builder(XContent xContent, Set<String> includes, S
 
     private boolean humanReadable = false;
 
+    private TimeUnit timeUnit = TimeUnit.MILLISECONDS;
```

Comment:
Yeah - it'd be super nice to be able to set the time unit on the URL and get it back - I'm quite happy at the idea. I'm just worried about the times when we persist stuff in XContent form and it lives for a while. We'd have to expand the places where we parse quite a bit. I think we could to it fairly easily for ObjectParser and friends but all the old places where we parse by hand are going to be a pain. They'll work now just because nothing ever changes this from millis.

That is why I propose moving it out of this - if you just did the names cleanup that'd be a good thing but supporting more units is going to be a bigger change.